src/hippopt/base/opti_solver.py

```diff
@@ -0,0 +1,400 @@
+                if not hasattr(corresponding_variable, field.name):
+                    raise ValueError(
+                        "The guess has the field "
+                        + field.name
+                        + " but it is not present in the optimization structure"
+                    )
+
+                self._set_initial_guess_internal(
+                    initial_guess=composite_variable_guess,
+                    corresponding_variable=corresponding_variable.__getattribute__(
+                        field.name
+                    ),
+                )
+                continue
+
+            is_list = isinstance(composite_variable_guess, list)
+            list_of_optimization_objects = is_list and all(
+                isinstance(elem, OptimizationObject)
+                for elem in composite_variable_guess
+            )
+
+            if list_of_optimization_objects:
+                if not hasattr(corresponding_variable, field.name):
+                    raise ValueError(
+                        "The guess has the field "
+                        + field.name
+                        + " but it is not present in the optimization structure"
+                    )
+                corresponding_nested_variable = corresponding_variable.__getattribute__(
+                    field.name
+                )
+
+                if not isinstance(corresponding_nested_variable, list):
+                    raise ValueError(
+                        "The guess has the field "
+                        + field.name
+                        + " as list, but the corresponding structure is not a list"
+                    )
+
+                i = 0
+                for element in composite_variable_guess:
+                    if i >= len(corresponding_nested_variable):
+                        raise ValueError(
+                            "The input guess is the list "
+                            + field.name
+                            + " but the corresponding variable structure is not a list"
+                        )
+
+                    self._set_initial_guess_internal(
+                        initial_guess=element,
+                        corresponding_variable=corresponding_nested_variable[i],
+                    )
+                    i += 1
+
+    def generate_optimization_objects(
+        self, input_structure: TOptimizationObject | List[TOptimizationObject], **kwargs
+    ) -> TOptimizationObject | List[TOptimizationObject]:
+        if isinstance(input_structure, OptimizationObject):
+            return self._generate_objects_from_instance(input_structure=input_structure)
+        return self._generate_objects_from_list(input_structure=input_structure)
+
+    def get_optimization_objects(
+        self,
+    ) -> TOptimizationObject | List[TOptimizationObject]:
+        return self._variables
+
+    def set_initial_guess(
+        self, initial_guess: TOptimizationObject | List[TOptimizationObject]
+    ):
```

Comment:
Just to understand, this only sets the attribute variables and not the paramenters, right?

Reply:
No, it sets both. You can skip the initialization of a given variable if you use ``None``

Reply:
Ok